dev/benchmarks/c/ipc_benchmark.cc

```diff
@@ -0,0 +1,178 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <stdio.h>
+
+#include <benchmark/benchmark.h>
+
+#include <nanoarrow/nanoarrow.hpp>
+#include <nanoarrow/nanoarrow_ipc.hpp>
+
+static ArrowErrorCode MakeFixtureInputStreamFile(const std::string& fixture_name,
+                                                 ArrowIpcInputStream* out) {
+  const char* fixture_dir = std::getenv("NANOARROW_BENCHMARK_FIXTURE_DIR");
+  if (fixture_dir == NULL) {
+    fixture_dir = "fixtures";
+  }
+
+  std::string fixture_path = std::string(fixture_dir) + std::string("/") + fixture_name;
+  FILE* fixture_file = fopen(fixture_path.c_str(), "rb");
+
+  NANOARROW_RETURN_NOT_OK(ArrowIpcInputStreamInitFile(out, fixture_file, true));
+  return NANOARROW_OK;
+}
+
+static ArrowErrorCode MakeFixtureBuffer(const std::string& fixture_name,
+                                        ArrowBuffer* out) {
+  nanoarrow::ipc::UniqueInputStream input_stream;
+  NANOARROW_RETURN_NOT_OK(MakeFixtureInputStreamFile(fixture_name, input_stream.get()));
+
+  nanoarrow::UniqueBuffer buffer;
+  int64_t size_read_out = 0;
+  int64_t chunk_size = 1024;
+  do {
+    NANOARROW_RETURN_NOT_OK(ArrowBufferReserve(buffer.get(), chunk_size));
+    NANOARROW_RETURN_NOT_OK(input_stream->read(input_stream.get(),
+                                               buffer->data + buffer->size_bytes,
+                                               chunk_size, &size_read_out, nullptr));
+    buffer->size_bytes += size_read_out;
+  } while (size_read_out > 0);
+
+  ArrowBufferMove(buffer.get(), out);
+  return NANOARROW_OK;
+}
+
+static ArrowErrorCode ArrayStreamReadAll(ArrowArrayStream* array_stream,
+                                         int64_t* batch_count, int64_t* column_count) {
+  nanoarrow::UniqueSchema schema;
+  NANOARROW_RETURN_NOT_OK(array_stream->get_schema(array_stream, schema.get()));
+  *column_count = schema->n_children;
+  benchmark::DoNotOptimize(schema);
+
+  nanoarrow::UniqueArrayView array_view;
+  NANOARROW_RETURN_NOT_OK(
+      ArrowArrayViewInitFromSchema(array_view.get(), schema.get(), nullptr));
+
+  while (true) {
+    nanoarrow::UniqueArray array;
+    NANOARROW_RETURN_NOT_OK(array_stream->get_next(array_stream, array.get()));
+    if (array->release == nullptr) {
+      break;
+    }
+
+    NANOARROW_RETURN_NOT_OK(
+        ArrowArrayViewSetArray(array_view.get(), array.get(), nullptr));
+
+    *batch_count = *batch_count + 1;
+  }
+
+  return NANOARROW_OK;
+}
+
+/// \defgroup nanoarrow-benchmark-ipc IPC Reader Benchmarks
+///
+/// Benchmarks for the ArrowArrayStream IPC reader.
+///
+/// @{
+
+/// \brief Use the ArrowArrayStream IPC reader to read 10,000 batches with 5 elements each
```

Comment:
So, does it make sense to benchmark tiny batches? This is a not a real-world situation at all.